### PR TITLE
Remove Lt.size in favor of Lt.slots_end pointer

### PR DESCRIPTION
**Pull request by travisstaloch**

### CODE REVIEW

**Comment by travisstaloch** [comment]
I believe I fixed the build issues.  Do I need to make another PR or is the commit I made automatically included?

File: src/system/lt.h [hunk 1]
```diff
@@ -46,19 +46,21 @@ static inline void *lt_push(Lt *lt, void *res, Dtor dtor)
...
-    if (lt->size >= lt->capacity) {
+    if ((size_t)(lt->slots_end - lt->slots) >= lt->capacity) {
         if (lt->capacity == 0) {
             lt->capacity = LT_INITIAL_CAPACITY;
             lt->slots = calloc(LT_INITIAL_CAPACITY, sizeof(Slot));
+            lt->slots_end = lt->slots;
         } else {
             lt->capacity *= 2;
             lt->slots = realloc(lt->slots, lt->capacity * sizeof(Slot));
+            lt->slots_end = lt->slots;
```
Comment (rexim) [inline src/system/lt.h]: You probably don't want to reset `slots_end` back to the beginning after the reallocation.

**Comment by travisstaloch** [comment]
I followed the changes you made on stream, setting slots_end to slots + size on realloc.  Also, I think the segfault you noticed happened because destroy_lt was attempting to access slots_end.  I changed the loop to start at slots_end - 1 and go to >= slots (instead of != slots).

**Comment by rexim** [comment]
@travisstaloch thank you very much for the contribution.

**Pull request merged**